code/game/objects/items/weapons/shards.dm

```diff
@@ -42,6 +40,10 @@
 	if(icon_prefix)
 		icon_state = "[icon_prefix][icon_state]"
 
+/obj/item/shard/Initialize()
+ . = ..()
+ initial_icon_state()
```

Comment:
```suggestion
 	. = ..()
 	AddComponent(/datum/component/caltrop, force)
 	initial_icon_state()
```
And move it back to this spot. Also, be sure to use tabs as indentation. I used spaces in my pseudo code because github is horrible with placing tabs 😅, but we use tabs in the code.